### ISSUE TITLE
TiKV is killed during profiling via TiDB-dashboard

### ISSUE THREAD

**Issue by overvenus** [tikv/tikv#9217]
### What version of TiKV are you using?

Master

### What operating system and CPU are you using?

Centos 7

### Steps to reproduce

Profile instance via dashboard.

```
$ less tikv.log
...
[2020/12/10 19:51:06.978 +08:00] [INFO] [sst_importer.rs:132] [download] [range="Some(start: 7480000000000000FF485F720380000000FF000005DC03800000FF000000BDE1000000FCFA2707D806DBFFFD end: 7480000000000000FF485F720380000000FF000005DE03800000FF0000013CC0000000FCFA2707D6A683F
FFE)"] [name=5_3401_678_9307d51b35bc51b33dfd88e22a6806de6d57ca82a02313ffb34ec5ac531370e1_1607583693309_default.sst] [meta="uuid: E115BA01D7CB45BFA14A56187D7EE423 range { start: 7480000000000000FF485F720380000000FF000005DC03800000FF000000BDE1000000FC end: 7480000000000000F
F485F720380000000FF000005DE03800000FF0000013CC1000000FC } length: 47128548 cf_name: \"default\" region_id: 24997 region_epoch { conf_ver: 8 version: 5192 }"]
[2020/12/10 19:51:06.979 +08:00] [INFO] [local.rs:31] ["create local storage"] [base=/data2/stn/backup/full3]
[2020/12/10 19:51:08.128 +08:00] [INFO] [profiler.rs:182] ["stopping cpu profiler"]
[2020/12/10 19:51:08.172 +08:00] [INFO] [mod.rs:401] ["write report successfully"]
[2020/12/10 19:51:08.250 +08:00] [WARN] [store.rs:712] ["[store 6] handle 72 pending peers include 72 ready, 0 entries, 102 messages and 0 snapshots"] [takes=1280]
[2020/12/10 19:51:26.563 +08:00] [INFO] [lib.rs:92] ["Welcome to TiKV"]
[2020/12/10 19:51:26.564 +08:00] [INFO] [lib.rs:97] ["Release Version:   4.1.0-alpha"]
[2020/12/10 19:51:26.564 +08:00] [INFO] [lib.rs:97] ["Edition:           Community"]
[2020/12/10 19:51:26.564 +08:00] [INFO] [lib.rs:97] ["Git Commit Hash:   520a69261628b3e7a6935a1c266b38045718d3e7"]
...

$ journalctl -u tikv-47906
...
Dec 10 19:37:41 ip-172-16-5-33 run_tikv.sh[35653]: user        0m0.000s
Dec 10 19:37:41 ip-172-16-5-33 run_tikv.sh[35653]: sys        0m2.316s
Dec 10 19:37:41 ip-172-16-5-33 run_tikv.sh[35653]: ok
Dec 10 19:51:08 ip-172-16-5-33 systemd[1]: tikv-47906.service: main process exited, code=killed, status=27/PROF
Dec 10 19:51:10 ip-172-16-5-33 systemd[1]: Unit tikv-47906.service entered failed state.
Dec 10 19:51:10 ip-172-16-5-33 systemd[1]: tikv-47906.service failed.
Dec 10 19:51:25 ip-172-16-5-33 systemd[1]: tikv-47906.service holdoff time over, scheduling restart.
Dec 10 19:51:25 ip-172-16-5-33 systemd[1]: Stopped tikv service.
Dec 10 19:51:25 ip-172-16-5-33 systemd[1]: Started tikv service.
Dec 10 19:51:26 ip-172-16-5-33 run_tikv.sh[34552]: sync ...
Dec 10 19:51:26 ip-172-16-5-33 run_tikv.sh[34552]: real        0m1.210s
Dec 10 19:51:26 ip-172-16-5-33 run_tikv.sh[34552]: user        0m0.000s
...
```

**Comment by YangKeao** [comment]
This problem is that: the timer still works after unregistering the signal handler. Then `tikv` received a `SIGPROF` and killed.

However, I have fixed this problem in `0.3.13`, which was published one year ago. And `tikv` used `pprof-rs 0.3.15` 🤦‍♂️ . I will try to reproduce this problem and find the reason.

**Comment by YangKeao** [comment]
Failed to reproduce on my low spec virtual machine with tpch load. I guess the problem is because of the signals which haven't arrived yet.
